src/SuperTokens.AspNetCore/Handshake.cs

```diff
@@ -1,31 +1,37 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
 
 namespace SuperTokens.AspNetCore
 {
     public sealed class Handshake
     {
+        private ImmutableArray<AccessTokenSigningKey> accessTokenSigningPublicKeyList;
+
         public Handshake(
-            string jwtSigningPublicKey,
-            DateTimeOffset jwtSigningPublicKeyExpiration,
+            IEnumerable<AccessTokenSigningKey> jwtSigningPublicKeyList,
             bool accessTokenBlacklistingEnabled,
             TimeSpan accessTokenLifetime,
             TimeSpan refreshTokenLifetime)
         {
-            this.JwtSigningPublicKey = jwtSigningPublicKey;
-            this.JwtSigningPublicKeyExpiration = jwtSigningPublicKeyExpiration;
+            this.AccessTokenSigningPublicKeyList = ImmutableArray.ToImmutableArray(jwtSigningPublicKeyList);
+
             this.AccessTokenBlacklistingEnabled = accessTokenBlacklistingEnabled;
             this.AccessTokenLifetime = accessTokenLifetime;
             this.RefreshTokenLifetime = refreshTokenLifetime;
         }
 
+        public ImmutableArray<AccessTokenSigningKey> AccessTokenSigningPublicKeyList
+        {
+            get => accessTokenSigningPublicKeyList.Where(keyInfo => keyInfo.Expiration > DateTimeOffset.Now).ToImmutableArray();
```

Comment:
I know I don't have anything in here test-wise just yet... but from a testability perspective, the dependency on `DateTimeOffset.Now` makes this impossible to run over a static data set (say in a regression test suite). It also injects the requirement that certain tests be dependent on a real clock advancing as opposed to a simulated clock.

Two possible refactorings could eliminate the use of `DateTimeOffset.Now`:

1. Turn this into a method that accepts a "current time" argument used to filter the list. The caller would then be responsible for using an abstracted clock (e.g. `Microsoft.AspNetCore.Authentication.ISystemClock`) to get the current time and feed it into the method.
2. Return the array as-is and let the caller apply the filtering logic as needed. The caller would still be responsible for using an abstracted clock (e.g. `Microsoft.AspNetCore.Authentication.ISystemClock`) to get the current time for use in its own logic.